app/src/main/kotlin/edu/stanford/bdh/engagehf/health/symptoms/SymptomsScreen.kt

```diff
@@ -190,12 +213,15 @@ fun SymptomsChart(
         shape = Shape.Pill,
     )
 
+    @Suppress("MagicNumber")
     val valueFormatter: (Float, ChartValues, AxisPosition.Vertical?) -> CharSequence =
-        { index, _, _ ->
-            val epochSecond = (index * EPOCH_SECONDS_DIVISOR).toLong()
-            val dateTime =
-                ZonedDateTime.ofInstant(Instant.ofEpochSecond(epochSecond), ZoneOffset.UTC)
-            dateTime.toLocalDate().format(DateTimeFormatter.ofPattern("MMM yy"))
+        { value, _, _ ->
```

Comment:
Missed one suggestion 😅 

Let's move this logic in the mapper. What we can do is the following. Extend the UiState object with a property: `val valueFormatter: (Float) -> String`

We would have something like this in the mapper:

```kotlin
    return SymptomsUiState.Success(
        data = SymptomsUiData(
            symptomScores = symptomScores,
            chartData = listOf(chartData),
            tableData = tableData,
            headerData = HeaderData(
                formattedValue = newestHealthData.formattedValue,
                formattedDate = newestHealthData.formattedDate,
                selectedSymptomType = selectedSymptomType,
                isSelectedSymptomTypeDropdownExpanded = false
            ),
            valueFormatter = ::valueFormatter
        )
    )
}
private fun valueFormatter(value: Float): String {
    val year = value.toInt()
    val dayOfYearFraction = value - year
    val dayOfYear = (dayOfYearFraction * 365).toInt() + 1
    val date = ZonedDateTime.of(year, 1, 1, 0, 0, 0, 0, ZoneId.systemDefault())
        .plusDays((dayOfYear - 1).toLong())
    return date.format(DateTimeFormatter.ofPattern("MMM dd"))
}
```

and here we can just have:

```kotlin
val valueFormatter: (Float, ChartValues, AxisPosition.Vertical?) -> CharSequence =
        { value, _, _ -> uiState.valueFormatter(value) }
```

It would be great to apply the same mechanism in HealthPages as well 🙏 This way we keep our composables dumb in terms of calculations, and increase the coverage in our unit tests :)